c2rust-analyze/src/borrowck/type_check.rs

```diff
@@ -234,6 +238,28 @@ impl<'tcx> TypeChecker<'tcx, '_> {
                     Some(Callee::Other { .. }) => {
                         // TODO
                     }
+                    Some(Callee::Malloc) => {
+                        // TODO
+                    }
+                    Some(Callee::Calloc) => {
+                        // TODO
+                    }
+                    Some(Callee::Realloc) => {
+                        // We handle this like a pointer assignment.
+                        let pl_lty = self.visit_place(destination);
+                        assert!(args.len() == 2);
+                        let rv_lty = self.visit_operand(&args[0]);
```

Comment:
Could we use [`assert_matches!`](https://docs.rs/assert_matches/latest/assert_matches/macro.assert_matches.html) to keep this in sync (the `.len()` `assert!` and the indexing)?

Reply:
added in 19fcf8ff